tests/integration/acp.go

```diff
@@ -107,12 +112,23 @@ func addPolicyACP(
 	s *state,
 	action AddPolicy,
 ) {
-	// If we expect an error, then ExpectedPolicyID should be empty.
-	if action.ExpectedError != "" && action.ExpectedPolicyID != "" {
+	// If we expect an error, then ExpectedPolicyID should never be provided.
+	if action.ExpectedError != "" && !action.ExpectedPolicyID.HasValue() {
 		require.Fail(s.t, "Expected error should not have an expected policyID with it.", s.testCase.Description)
 	}
 
 	nodeIDs, nodes := getNodesWithIDs(action.NodeID, s.nodes)
+	maxNodeID := slices.Max(nodeIDs)
+	policyIDsAddedSoFar := len(s.policyIDs)
+	// Expand the policyIDs slice once, so we can minimize how many times we need to expaind it.
+	// We use the maximum nodeID provided to make sure policyIDs slice can accomodate upto that nodeID.
+	if policyIDsAddedSoFar <= maxNodeID {
```

Comment:
nitpick: would prefer `if len(s.policyIDs) <= maxNodeID` as the variable doesn't add any clarity.